doc/example/limitations_template_method.cpp

```diff
@@ -36,6 +34,14 @@ MOCK_CLASS( mock_concept )
     MOCK_METHOD( method, 1, void( const char* ), method_string )
 };
 //]
+
+BOOST_AUTO_TEST_CASE(mocked_templated_methods_are_called)
+{
+    mock_concept b;
+    MOCK_EXPECT(b.method_int).once().with(42);
+    MOCK_EXPECT(b.method_string).once().with(mock::equal(std::string("string")));
```

Comment:
`mock::equal("string")` should work, right?

Reply:
The method takes a `const char*` so `mock::equal("string")` would compare pointers, not the values. The conversion to string makes it compare the actual strings instead

Reply:
Oh yeah, there used to be an overload for `mock::equal` with a `const char*` to act as a special case to compare strings, but this was removed in a later version of turtle 😬 
Maybe
```suggestion
    MOCK_EXPECT(b.method_string).once().with(mock::equal<std::string>("string"));
```
though?

Reply:
I don't see any advantage here. Passing a string is IMO more clear than changing template params

Reply:
I still think `mock::equal<std::string>` is a bit clearer, as in meaning «equal as a string», but that’s pretty minor.